synapse/handlers/federation.py

```diff
@@ -946,6 +946,12 @@ async def on_invite_request(
         )
 
         context = EventContext.for_outlier(self._storage_controllers)
+
+        if event.is_notifiable:
+            await self._federation_event_handler.bulk_push_rule_evaluator.action_for_event_by_user(
+                event, context
+            )
+
```

Comment:
We'll want to call `remove_push_actions_from_staging` when an exception happens in `persist_events_and_notify`, just like in https://github.com/matrix-org/synapse/blob/0bf180cbb43f6e3d489ddaa5d66226b6daeea027/synapse/handlers/federation_event.py#L2117-L2123

Reply:
Ooof, thanks for noticing! I saw that in the other persist call and thought about putting it here as well, but forgot.